File: jsol-compiler-src/cli-parser.py
import math
from jsol_core import JSOL

# @JSOL v0.2.95 - CLI Arguments Parser
def _mParseRawCliArgs(_aRawArgs): 

  _mOptions = JSOL.dict(
  "source",  "", 
  "outDir",  "", 
  "target",  "", 
  "jsTarget",  "", 
  "jsPrefix",  "", 
  "jsSuffix",  "", 
  "phpTarget",  "", 
  "phpPrefix",  "", 
  "phpSuffix",  "", 
  "tsTarget",  "", 
  "tsPrefix",  "", 
  "tsSuffix",  "", 
  "pyTarget",  "", 
  "pyPrefix",  "", 
  "pySuffix",  "", 
  "targets",  ""
  );

  _iCount = len(_aRawArgs);
  _i = 0;
  while _i < _iCount: 

    _sArg = _aRawArgs[_i];
    _bIsFlag = (JSOL.str_index_of(_sArg,  "--") == 0);

    if _bIsFlag == True: 

      _sClean = _sArg[( 2):( 2)+( len(_sArg) - 2)];
      _iEqIndex = JSOL.str_index_of(_sClean,  "=");
      _sKey = "";
      _sVal = "";

      if _iEqIndex != -1: 

        _sKey = _sClean[( 0):( 0)+( _iEqIndex)];
        _sVal = _sClean[( _iEqIndex + 1):( _iEqIndex + 1)+( len(_sClean) - (_iEqIndex + 1))];


      else: 

        _sKey = _sClean;
        _sVal = "true";


      if _sKey == "source": 

        _mOptions["source"] = _sVal;


      if _sKey == "out-dir": 

        _mOptions["outDir"] = _sVal;


      if _sKey == "targets": 

        _mOptions["targets"] = _sVal;


      if _sKey == "target": 

        _mOptions["target"] = _sVal;
        _mOptions["jsTarget"] = _sVal;
        _mOptions["phpTarget"] = _sVal;
        _mOptions["tsTarget"] = _sVal;
        _mOptions["pyTarget"] = _sVal;


      if _sKey == "js-target": 

        _mOptions["jsTarget"] = _sVal;


      if _sKey == "js-prefix": 

        _mOptions["jsPrefix"] = _sVal;


      if _sKey == "js-suffix": 

        _mOptions["jsSuffix"] = _sVal;


      if _sKey == "php-target": 

        _mOptions["phpTarget"] = _sVal;


      if _sKey == "php-prefix": 

        _mOptions["phpPrefix"] = _sVal;


      if _sKey == "php-suffix": 

        _mOptions["phpSuffix"] = _sVal;


      if _sKey == "ts-target": 

        _mOptions["tsTarget"] = _sVal;


      if _sKey == "ts-prefix": 

        _mOptions["tsPrefix"] = _sVal;


      if _sKey == "ts-suffix": 

        _mOptions["tsSuffix"] = _sVal;


      if _sKey == "py-target": 

        _mOptions["pyTarget"] = _sVal;


      if _sKey == "py-prefix": 

        _mOptions["pyPrefix"] = _sVal;


      if _sKey == "py-suffix": 

        _mOptions["pySuffix"] = _sVal;




    _i = _i + 1;


  return _mOptions;

File: jsol-compiler-src/cli-parser.js
// @JSOL v0.2.95 - CLI Arguments Parser
const $mParseRawCliArgs = function($aRawArgs) {
  const $mOptions = JSOL.dict("source",  "",  "outDir",  "",  "target",  "",  "jsTarget",  "",  "jsPrefix",  "",  "jsSuffix",  "",  "phpTarget",  "",  "phpPrefix",  "",  "phpSuffix",  "",  "tsTarget",  "",  "tsPrefix",  "",  "tsSuffix",  "",  "pyTarget",  "",  "pyPrefix",  "",  "pySuffix",  "",  "targets",  "");
	
    const $iCount = $aRawArgs.length;
    for (let $i = 0; $i < $iCount; $i = $i + 1) {
    const $sArg = $aRawArgs[$i];
        let $bIsFlag = false;
         $bIsFlag = $sArg.indexOf("--") === 0; if ($bIsFlag === true) {
      let $sKey = ""; let $sVal = "";
            
                const $sClean = $sArg.substring(2);
                const $iEqIndex = $sClean.indexOf("=");
                if ($iEqIndex !== -1) {
        $sKey = $sClean.substring(0, $iEqIndex); $sVal = $sClean.substring($iEqIndex + 1);
      }
      else {
        $sKey = $sClean; $sVal = "true";
      }
      if ($sKey === "source") {
        $mOptions["source"] = $sVal;
      }
      if ($sKey === "out-dir") {
        $mOptions["outDir"] = $sVal;
      }
      if ($sKey === "targets") {
        $mOptions["targets"] = $sVal;
      }
      if ($sKey === "target") {
        $mOptions["target"] = $sVal; $mOptions["jsTarget"] = $sVal; $mOptions["phpTarget"] = $sVal; $mOptions["tsTarget"] = $sVal; $mOptions["pyTarget"] = $sVal;
      }
      if ($sKey === "js-target") {
        $mOptions["jsTarget"] = $sVal;
      }
      if ($sKey === "js-prefix") {
        $mOptions["jsPrefix"] = $sVal;
      }
      if ($sKey === "js-suffix") {
        $mOptions["jsSuffix"] = $sVal;
      }
      if ($sKey === "php-target") {
        $mOptions["phpTarget"] = $sVal;
      }
      if ($sKey === "php-prefix") {
        $mOptions["phpPrefix"] = $sVal;
      }
      if ($sKey === "php-suffix") {
        $mOptions["phpSuffix"] = $sVal;
      }
      if ($sKey === "ts-target") {
        $mOptions["tsTarget"] = $sVal;
      }
      if ($sKey === "ts-prefix") {
        $mOptions["tsPrefix"] = $sVal;
      }
      if ($sKey === "ts-suffix") {
        $mOptions["tsSuffix"] = $sVal;
      }
      if ($sKey === "py-target") {
        $mOptions["pyTarget"] = $sVal;
      }
      if ($sKey === "py-prefix") {
        $mOptions["pyPrefix"] = $sVal;
      }
      if ($sKey === "py-suffix") {
        $mOptions["pySuffix"] = $sVal;
      }
    }
  }
  return $mOptions;
};
